Release SporTran `v1.0.0rc2`

lorisercole opened the pull request:
Following the comments of the Computer Physics Communications' reviewer, we implemented the following changes:

- [x] Define `Current` as an abstract class (closes #59, see 902b2f61e7dc797a299e06c8e2bb19c59020991d)
  - The definition of the `_builder` method, `_current_type`, `_input_parameters`, and `_KAPPA_SI_UNITS` must be implemented in a subclass. 
  - A `GenericCurrent` (without units, where only `KAPPA_SCALE` is defined) has been defined. This takes the place of `Current`.

- [x] LAMMPS examples added (closes #18, closes #23)
  - [x] LAMMPS input/output files for silica (be12473)
  - [x] script to convert LAMMPS output file to `table` (plain text) and `dict` (numpy binary) format (be12473)
  - [x] example showing how to use the `TableFile` and `LAMMPSLogFile` classes from the `i_o` module (a7a4657)
  - [ ] LAMMPS input/output files for NaCl ? (@rikigigi)

- [x] Clarify the use of units: add appropriate definitions of the currents (formulas) in the documentation (@rikigigi)
  - [x] Delete units that are not correct or available.
- [x] Review list of requirements and package metadata.

lorisercole committed:
Update pre-commit config

- yapf v0.32.0
- pre-commit-hooks v4.1.0

lorisercole committed:
Move `fstar_analysis` to current/tools

lorisercole committed:
Define `Current` abstract class, and `GenericCurrent`

Solves #59
`Current` takes the role of an abstract class.
The `_current_type`, `_input_parameters`, and `_KAPPA_SI_UNITS`
attributes, and the `_builder` property method must be defined in
a derived subclass.

lorisercole committed:
Get rid of `dlpoly` units

Heat/electric currents in the DLPOLY code are not yet publicly available,
hence they are not useful and may create confusion.

lorisercole committed:
Move example data into examples/data subfolders

lorisercole committed:
Improve read_lammps_log

- Fixed the binary writer
- added --lammps-structure-file as cli input
- make sure that all columns are read if no column is selected

lorisercole committed:
Add LAMMPS Silica input/output files

Fixes #18
Contributes to #23
The LAMMPS input files used to generate the silica example data have
been added to the `examples/data/Silica/lammps` folder.
In the same folder, the output of the calculation is reported, along
with a script to convert it to `table` and `dict` formats.
The data had to be regenerated, therefore all the data files in the
example folder have been updated.

lorisercole committed:
Fix tests, add script to regenerate data

Tests using Silica data had to be updated after commit be124735d893899dd018f5ebbbf9bbc3da5d94b3
We also added a script to regenerate the data in the format required by
tests, for convenience.

lorisercole committed:
Rename examples, add silica cmdline example

lorisercole committed:
Fix TableFile bug

If EOF was reached, the number of steps NSTEPS was not updated and the
initialized data vectors were not resized.
Now it is fixed.

lorisercole committed:
read_lammps_log docs and variable change

- The documentation has been updated.
- `lammps_structure_file` has been renamed `lammps_data_file`
- `get_box` function is imported from the `read_lammps_datafile` module

lorisercole committed:
Reorganize `i_o` folder

binary.py moved to `_sportran_binary`
It is a module still under development.

lorisercole committed:
Add input formats example

and fix `convert_lammps_log.sh` script

lorisercole committed:
Fix notebooks test

rikigigi committed:
add test for new notebook, updated symlinks

rikigigi committed:
little update of the documentation

rikigigi committed:
update of the units documentation.
Removal of unused or wrong conversion factors

rikigigi committed:
added small description of folder with lammps example; added jupyter notebook with inputs format example also in docs

lorisercole committed:
Fix small typos in documentation

lorisercole commented on sportran/current/units/stress.py:
```diff
@@ -10,51 +10,30 @@ def scale_kappa_GPa(TEMPERATURE, VOLUME):
+    Input stress in atmospheres.
     """
-    return constants.atm**2 * 1.0e-12 / TEMPERATURE / constants.kB / VOLUME
+    return constants.atm**2 * 1.0e-12 * VOLUME / TEMPERATURE / constants.kB
```
@fgrassel Does this look correct to you?

fgrassel replied:
Yes, with the link to the LAMMPS documentation and the remark that the input is in atmospheres, this looks fine.

lorisercole commented on sportran/current/units/stress.py:
```diff
@@ -10,51 +10,30 @@ def scale_kappa_GPa(TEMPERATURE, VOLUME):
-    VOLUME      cell VOLUME [A^3]
-    """
-    return constants.charge**2 * VOLUME / TEMPERATURE / constants.kB
+    return 1.0e-12 * VOLUME / TEMPERATURE / constants.kB
```
@fgrassel same question for this one

fgrassel replied:
Yes, it is fine.

lorisercole commented:
All good, but some comments on the units. Let's wait for @fgrassel review!

lorisercole committed:
Fix label in GUI header selection

'Energy current' changed to 'Main current'.

rikigigi committed:
restored some units that were wrongly removed

lorisercole committed:
Update metadata, dependencies, and package data

- The plot style files have been added to the package data, this fixes and
error that can happen in certain installations, for which these files
cannot be retrieved because they are not installed.
- Package metadata has been updated: development status is now
"Production/Stable", and other categories added.
- Some dependencies have been bumped. New python version may still have
problems to install the minimum deps versions.

lorisercole committed:
Add Python 3.9, 3.10 support & tests
